### PR TITLE
Connect Angular frontend to Express/Lambda backend

### LINKED ISSUE #352: Connect the Angular front end to the Lambda backend
**Background**

In the new tech stack, the front-end application needs to communicate with the Lambda backend. Once we have finished #351 and #350 and deployed the Angular application on AWS then we should configure the Angular application to call the Lambda API and display something from the API call in the application.

**We'll know we're done when...**
- [ ] We can load our Angular application from AWS and see the resulting information that came from the Lambda API.

**Tasks**
- [ ] Research how and set up the Angular application to deploy on AWS
- [ ] Configure the Angular application with the location of the Lambda API
- [ ] Make an API call in Angular and display the result of the API call on the page

### PR DESCRIPTION
**Pull request by beechnut**
- Adds a service to the Angular app that connects to the Express API
- Adds CORS headers to Express

Closes #352

### CODE REVIEW

File: api/src/index.ts [hunk 1]
```diff
@@ -3,6 +3,13 @@ import express, { Request, Response, NextFunction } from 'express';
 const app  = express();
 const port = 3000;
 
+app.use(function(req, res, next) {
+  res.setHeader('Access-Control-Allow-Origin', 'http://localhost:4200')
```
Comment (neilmb) [inline api/src/index.ts]: **question:** If you are adding per-environment controls for Angular in this PR, do we need to do something similar for the API to listen on a safe URL? Or is this part of the Express infrastructure that isn't run on Lambda? If so, maybe big, loud comments in this file about what is only part of the application for local development?

File: frontend/src/app/app.module.ts [hunk 2]
```diff
@@ -2,13 +2,18 @@ import { NgModule } from '@angular/core';
 import { BrowserModule } from '@angular/platform-browser';
 
 import { AppComponent } from './app.component';
+import { TimezonesComponent } from './timezones/timezones.component';
+
+import { HttpClientModule } from '@angular/common/http';
```
Comment (neilmb) [inline frontend/src/app/app.module.ts]: **nitpick:** Is there a standard for how these imports are ordered in TS? I would expect to see this one with the other `@angular/...` import above the local imports.

File: frontend/src/app/timezone.service.spec.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+describe('TimezoneService', () => {
+  let service: TimezoneService;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({});
+    service = TestBed.inject(TimezoneService);
+  });
+
+  it('should be created', () => {
```
Comment (neilmb) [inline frontend/src/app/timezone.service.spec.ts]: **question (if-minor):** Do we have documentation in our README on how to run these tests during local development? And are they running now in CI, or is that part of the planning work that we are still doing? If not, would adding the running of these tests to the existing setup be worth doing in this PR?

File: frontend/src/app/timezone.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+export interface Timezone {
```
Comment (neilmb) [inline frontend/src/app/timezone.ts]: **question:** There is a very similar typedef in https://github.com/USDAForestService/NRM-Grants-Agreements/blob/development/api/src/index.ts#L10. For this, I'm fine repeating it here, but it has me wondering about re-use between front- and backend code in the future.

**Review by neilmb — DISMISSED** [review]
LGTM. Lots of questions, but nothing that needs to stand in the way.

File: frontend/src/app/app.module.ts (hunk 2 above)
Comment (beechnut) [reply frontend/src/app/app.module.ts]: I just added Prettier with a plugin that organizes TypeScript imports. Before I commit and push it, do you have an opinion on how you'd want Prettier to run? I'm thinking pre-commit hook, but I'm open to other ideas.

File: frontend/src/app/timezone.service.spec.ts (hunk 3 above)
Comment (beechnut) [reply frontend/src/app/timezone.service.spec.ts]: @neilmb Glad you're paying attention to this. I do think this is best done as part of another story. This one is just to connect the apps, but perhaps #289 would make a good home for this (even though they're unit tests).

File: frontend/src/app/timezone.ts (hunk 4 above)
Comment (beechnut) [reply frontend/src/app/timezone.ts]: That...is a great question and definitely worth exploring. I opened #373 to dig into this further.

File: api/src/index.ts (hunk 1 above)
Comment (beechnut) [reply api/src/index.ts]: Trying to reply to this specific question had me realize that while the frontend and backend are connected in that they know how to locate each other locally, this PR does not yet cover deploying Angular at a path through the Express app. This seems like an essential part of this story, so I'll continue working on that.

Comment (beechnut) [reply api/src/index.ts]: I ended up having to set CORS headers to `'*'` to get the sandbox deployment to work, and I'm not sure why. Setting it to the frontend host (the S3 URL) didn't work. I wonder if this has something to do with the API Gateway, but the resources we're using via Claudia aren't visible to me in the sandbox environment.

**Review by neilmb — APPROVED** [review]
LGTM. Glad to see Prettier at work in the source code and worked well locally with `docker-compose up`.

**Comment by jayrbergjr** [comment]
@beechnut I approve, I can't figure out how to check my request.  :(

**Review by jayrbergjr — APPROVED** [review]

**Pull request merged**